pkg/operator/metrics/metrics.go

```diff
@@ -0,0 +1,161 @@
+package metrics
+
+import (
+	"context"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	log "github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+	"github.com/openshift/hive/pkg/constants"
+)
+
+const (
+	ControllerName = hivev1.MetricsControllerName
+)
+
+var (
+	// Capture the various conditions set on the HiveConfig
+	metricHiveConfigConditions = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "hive_config_conditions",
+		Help: "HiveConfig with conditions.",
+	}, []string{"condition", "reason"})
```

Comment:
I am legit not sure if reason should be here. How does this behave if the reason is changing? 
It doesn't feel like a cardinality problem.
I imagine the graph will just be different lines at value 1 over time, and the time series might change in the middle.

I *think* we can still do an alert that triggers if a condition is True for 10 minutes, even if the Reason changes mid-way right? We'd just be querying only on Condition right?

Reply:
I don't think hiveconfig is complicated enough to include the reason - although we could if we want. But yes - the value will change everytime the reason changes even if the status doesn't.

Reply:
I was thinking of having this as a table showing condition, reason and status (more on the lines of `up` metric) instead of time series graph. 

For alerting, yes we would check on condition and status, doesn't matter if reason changes midway.

Reply:
It's definitely useful in a table view, and if everything works ok for an alert across a timespan when we're potentially hitting multiple time series, then I'm ok with including reason.